File: php/lc00001-lc00100/lc00004/MedianOfTwoSortedArraysTest.php
<?php

namespace leetcode\lc00004;

use PHPUnit\Framework\TestCase;

class MedianOfTwoSortedArraysTest extends TestCase
{
    public function testFindMedianSortedArrays()
    {
        $cases = [
            ["nums1" => [1, 3], "nums2" => [2], "expected" => 2],
            ["nums1" => [1, 2], "nums2" => [3, 4], "expected" => 2.5],
            ["nums1" => [0, 0], "nums2" => [0, 0], "expected" => 0],
            ["nums1" => [], "nums2" => [1], "expected" => 1],
            ["nums1" => [2], "nums2" => [], "expected" => 2],
        ];

        $obj = new MedianOfTwoSortedArrays();
        for ($i = 0; $i < count($cases); $i++) {
            $this->assertEquals($cases[$i]['expected'], $obj->findMedianSortedArrays($cases[$i]['nums1'], $cases[$i]['nums2']));
        }
    }
}

File: golang/lc00001-lc00100/lc00004/median_of_two_sorted_arrays_test.go
package lc00004

import (
	"testing"
)

func TestFindMedianSortedArrays(t *testing.T) {
	type caseType struct {
		nums1    []int
		nums2    []int
		excepted float64
	}

	cases := []caseType{
		{[]int{1, 3}, []int{2}, 2},
		{[]int{1, 2}, []int{3, 4}, 2.5},
		{[]int{0, 0}, []int{0, 0}, 0},
		{[]int{}, []int{1}, 1},
		{[]int{2}, []int{}, 2},
	}

	for _, c := range cases {
		result := findMedianSortedArrays(c.nums1, c.nums2)
		if result != c.excepted {
			t.Fatalf("findMedianSortedArrays function failed, nums1: %v, nums2: %v, execpted: %f, result: %f", c.nums1, c.nums2, c.excepted, result)
		}
	}
}
